### PR TITLE
[SYCL][JointMatrix] New tests for arrays of Joint Matrix

### PR DESCRIPTION
**Pull request by YuriPlyakhin**
New tests for arrays of Joint Matrix

### CODE REVIEW

**Review by dkhaldi — COMMENTED** [review]
can we add the DG2/ATS-M version so we can test the fix there too? (SG size = 8)

**Comment by YuriPlyakhin** [comment]
> can we add the DG2/ATS-M version so we can test the fix there too? (SG size = 8)

sure, this is in my list, did not have time yet.

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+           // For B, we assume B has been already VNNIed.
+           joint_matrix<sub_group, bfloat16, use::b, TK, TN,
+                        ext::intel::experimental::matrix::layout::packed>
+               sub_b;
+           joint_matrix<sub_group, float, use::accumulator, TM, TN>
+               sub_c[JM_ARRAY_SZ];
+
+           for (int i = 0; i < JM_ARRAY_SZ; ++i)
+             joint_matrix_fill(sg, sub_c[i], 1.0);
```
[the remark this answers is not in the document]
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp:64]: I run these tests on ATSM in RIL, they were passing. I can run once again, need to setup ATSM in RIL.

[the remark this answers is not in the document]
Comment (dkhaldi) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp:64]: It is possible that they added emulation for fp64

**Review by dkhaldi — APPROVED** [review]

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp:64]: Just verified one more time, it passes on ATS-M:
icpx -fsycl /llvm/sycl/test-e2e/Matrix/XMX8/joint_matrix_bfloat16_array.cpp
./a.out 
passed

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by YuriPlyakhin** [comment]
@intel/llvm-reviewers-runtime , @sergey-semenov , could you please approve?

**Review by aelovikov-intel — APPROVED** [review]

File: sycl/test-e2e/Matrix/XMX8/joint_matrix_bf16_fill_k_cache.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+using namespace sycl;
+using namespace sycl::ext::intel;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using bfloat16 = sycl::ext::oneapi::bfloat16;
```
[the remark this answers is not in the document]
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/XMX8/joint_matrix_bf16_fill_k_cache.cpp]: done

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , please, merge

**Pull request merged**